RecoTracker/LSTCore/src/alpaka/MiniDoublet.h

```diff
@@ -0,0 +1,1093 @@
+          (side == PosZ and layer == 1 and rod < 4))
+        return true;
+      else
+        return false;
+    } else
+      return false;
+  };
+
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE float moduleGapSize(struct SDL::modules& modulesInGPU, uint16_t& moduleIndex) {
+    float miniDeltaTilted[3] = {0.26f, 0.26f, 0.26f};
+    float miniDeltaFlat[6] = {0.26f, 0.16f, 0.16f, 0.18f, 0.18f, 0.18f};
+    float miniDeltaLooseTilted[3] = {0.4f, 0.4f, 0.4f};
+    float miniDeltaEndcap[5][15];
+
+    for (size_t i = 0; i < 5; i++) {
+      for (size_t j = 0; j < 15; j++) {
+        if (i == 0 || i == 1) {
+          if (j < 10) {
+            miniDeltaEndcap[i][j] = 0.4f;
+          } else {
+            miniDeltaEndcap[i][j] = 0.18f;
+          }
+        } else if (i == 2 || i == 3) {
+          if (j < 8) {
+            miniDeltaEndcap[i][j] = 0.4f;
+          } else {
+            miniDeltaEndcap[i][j] = 0.18f;
+          }
+        } else {
+          if (j < 9) {
+            miniDeltaEndcap[i][j] = 0.4f;
+          } else {
+            miniDeltaEndcap[i][j] = 0.18f;
+          }
+        }
+      }
+    }
+
+    unsigned int iL = modulesInGPU.layers[moduleIndex] - 1;
+    unsigned int iR = modulesInGPU.rings[moduleIndex] - 1;
+    short subdet = modulesInGPU.subdets[moduleIndex];
+    short side = modulesInGPU.sides[moduleIndex];
+
+    float moduleSeparation = 0;
+
+    if (subdet == Barrel and side == Center) {
+      moduleSeparation = miniDeltaFlat[iL];
+    } else if (isTighterTiltedModules(modulesInGPU, moduleIndex)) {
+      moduleSeparation = miniDeltaTilted[iL];
+    } else if (subdet == Endcap) {
+      moduleSeparation = miniDeltaEndcap[iL][iR];
+    } else  //Loose tilted modules
+    {
+      moduleSeparation = miniDeltaLooseTilted[iL];
+    }
+
+    return moduleSeparation;
+  };
+
+  template <typename TAcc>
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE float dPhiThreshold(TAcc const& acc,
+                                                     float rt,
+                                                     struct SDL::modules& modulesInGPU,
+                                                     uint16_t& moduleIndex,
+                                                     float dPhi = 0,
+                                                     float dz = 0) {
+    // =================================================================
+    // Various constants
+    // =================================================================
+    //mean of the horizontal layer position in y; treat this as R below
+
+    // =================================================================
+    // Computing some components that make up the cut threshold
+    // =================================================================
+
+    unsigned int iL = modulesInGPU.layers[moduleIndex] - 1;
+    const float miniSlope = alpaka::math::asin(acc, alpaka::math::min(acc, rt * k2Rinv1GeVf / ptCut, sinAlphaMax));
```

Comment:
I am working on removing the `ALPAKA_STATIC_ACC_MEM_GLOBAL` keyword from our `Constants.h` file. When I remove it from in front of `sinAlphaMax`, I get:
```
src/RecoTracker/LSTCore/src/alpaka/MiniDoublet.h(358): error: identifier "lst::sinAlphaMax" is undefined in device code
```

However, if I do:
```suggestion
    const float miniSlope = alpaka::math::asin(acc, alpaka::math::min(acc, rt * k2Rinv1GeVf / ptCut, 1.f * sinAlphaMax));
```
it compiles just fine. Any idea what is going on? If not, I can propagate the comment to the CMSSW PR.

The same happens with other variables.

Reply:
The function expects a reference (`T const&`) in the arguments. 
In the example above with `1.f*` the compiler generates a temporary first, which works.
But when you use it directly, the constexpr (without static specifiers) has no address.

Reply:
I see, it makes sense. How should we handle it then? Should I just keep the `ALPAKA_STATIC_ACC_MEM_GLOBAL` for the constants that we use "alone" or create temporaries for those and get rid of the keyword to have a consistent treatment for the constants?

Reply:
This seems to be a guideline topic; better ask Matti.